vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -25,12 +25,35 @@ window.Vaadin.Flow.comboBoxConnector = {
 
     comboBox.$connector = {};
 
+    // holds pageIndex -> callback pairs of subsequent indexes (current active range)
     let pageCallbacks = {};
     let cache = {};
     let lastFilter = '';
+    const placeHolder = new Vaadin.ComboBoxPlaceholder();
+    const MAX_RANGE_COUNT = Math.max(comboBox.pageSize * 2, 150); // Max item count in active range
```

Comment:
This could also be increased, e.g. to 300 or 500 to avoid re-loading data too much (like you suggested yourself IIRC).

Actually I'm not even sure anymore if this is needed with the current implementation. What was the reason again to dump old pages also when not skipping pages? 😅 Well, it works the same way as Grid, so probably fine this way.

Reply:
> What was the reason again to dump old pages also when not skipping pages?

If you use one of those super mice you have, you can quickly scroll down and buffer a request of thousands of items before it's sent. Increased to 500.

Reply:
Right 👍